hangman.py

```diff
@@ -0,0 +1,330 @@
+    It returns false or true, which is understood as loose or win in hangman.
+    '''
+    while True:
+        print("--------------------------------------")
+        if guesses_left <= 0:
+            return False
+        guessed_letter = output_information(guesses_left, letters_guessed)
+        if hints_switch:
+            #Game with hints activated
+            if guessed_letter == SYMBOL_FOR_HINTS:
+                show_possible_matches(get_guessed_word(secret_word, letters_guessed))
+                continue
+        if guessed_letter.isalpha():
+            guessed_letter = guessed_letter.lower()
+        else:
+            #System of losing warnings/guesses activated
+            if warnings_left == 0:
+                guesses_left = not_correct_input(warnings_left, guesses_left, secret_word, letters_guessed)
+            else:
+                warnings_left = not_correct_input(warnings_left, guesses_left, secret_word, letters_guessed)
+            continue
+        if guessed_letter in letters_guessed:
+            #System of losing warnings/guesses activated
+            warnings_left = guessed_before(warnings_left, guesses_left, letters_guessed)
+            if warnings_left == 0:
+                guesses_left = guessed_before(warnings_left, guesses_left, letters_guessed)
+            else:
+                warnings_left = guessed_before(warnings_left, guesses_left, letters_guessed)
+            continue
+        else:
+            letters_guessed.append(guessed_letter)
+            guesses_left = results(guessed_letter, guesses_left, letters_in_secret_word, secret_word, letters_guessed)
+        if is_word_guessed(secret_word, letters_guessed):
+            return True
+
+def hangman(secret_word, hints_switch = False):
+    '''
+    secret_word: string, the secret word to guess.
+
+    Starts up an interactive game of Hangman.
+
+    * At the start of the game, let the user know how many
+      letters the secret_word contains and how many guesses s/he starts with.
+
+    * The user should start with 6 guesses
+
+    * Before each round, you should display to the user how many guesses
+      s/he has left and the letters that the user has not yet guessed.
+
+    * Ask the user to supply one guess per round. Remember to make
+      sure that the user puts in a letter!
+
+    * The user should receive feedback immediately after each guess
+      about whether their guess appears in the computer's word.
+
+    * After each guess, you should display to the user the
+      partially guessed word so far.
+
+    Follows the other limitations detailed in the problem write-up.
+    '''
+    #Instruction to the game for player
+    letters_in_secret_word = set(secret_word)
+    print("Welcome to the game Hangman!")
+    print("I am thinking of a word that is", len(secret_word), "letters long.")
+    print("You have 3 warnings left")
+    guesses_left = GUESSES_AT_START
+    warnings_left = WARNINGS_AT_START
+    letters_guessed = []
```

Comment:
Do you think that list has the most effectiveness in that case?

Reply:
done